appcelerator-archive/amp
# Gateway HTTP logger

Adding HTTP logs (combined format) to `amplifier-gateway` in order to trace HTTP activity.

## How to test
```
$ ampmake build
$ amp cluster create -t local
[Using the UI, log in and display logs and metrics]
$ amp logs -i amp_amplifier-gateway
[You should see HTTP activity, for instance:]
10.0.0.25 - - [06/Jun/2017:23:11:46 +0000] "POST /v1/stats HTTP/1.1" 200 1984 "https://local.appcelerator.io/amp/dashboard" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36"
10.0.0.25 - - [06/Jun/2017:23:11:46 +0000] "POST /v1/stats HTTP/1.1" 200 6381 "https://local.appcelerator.io/amp/dashboard" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36"
10.0.0.25 - - [06/Jun/2017:23:11:46 +0000] "POST /v1/stats HTTP/1.1" 200 1984 "https://local.appcelerator.io/amp/dashboard" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36"
10.0.0.25 - - [06/Jun/2017:23:11:46 +0000] "POST /v1/stats HTTP/1.1" 200 1984 "https://local.appcelerator.io/amp/dashboard" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36"
10.0.0.25 - - [06/Jun/2017:23:11:46 +0000] "POST /v1/stats HTTP/1.1" 200 1984 "https://local.appcelerator.io/amp/dashboard" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36"
10.0.0.25 - - [06/Jun/2017:23:11:50 +0000] "OPTIONS /v1/stacks HTTP/1.1" 200 0 "https://local.appcelerator.io/amp/stacks" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36"
10.0.0.25 - - [06/Jun/2017:23:11:50 +0000] "GET /v1/stacks HTTP/1.1" 200 2 "https://local.appcelerator.io/amp/stacks" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36"
10.0.0.25 - - [06/Jun/2017:23:11:51 +0000] "OPTIONS /v1/logs HTTP/1.1" 200 0 "https://local.appcelerator.io/amp/logs" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36"
10.0.0.25 - - [06/Jun/2017:23:11:51 +0000] "POST /v1/logs HTTP/1.1" 200 10847 "https://local.appcelerator.io/amp/logs" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36"
10.0.0.25 - - [06/Jun/2017:23:11:51 +0000] "OPTIONS /v1/stats HTTP/1.1" 200 0 "https://local.appcelerator.io/amp/metrics/global/single/all" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36"
10.0.0.25 - - [06/Jun/2017:23:11:51 +0000] "POST /v1/stats HTTP/1.1" 200 5919 "https://local.appcelerator.io/amp/metrics/global/single/all" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36"
```
